frontend/src/components/useLocalStorage.ts

```diff
@@ -0,0 +1,50 @@
+import { useState } from "react";
+
+type Setter<T> = (
+  value: T | ((oldValue: T) => T),
+  updateStorage: boolean
+) => void;
+
+// Create a state in the same way as `useState` while binding the state with
+// localStorage. All stored objects will be serialized with JSON.stringify
+export function useLocalStorage<T>(
+  key: string,
+  initialValue: T
+): [T, Setter<T>] {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === "undefined") {
+      return initialValue;
+    }
+    try {
+      // Get from local storage by key
+      const item = window.localStorage.getItem(key);
+      // Parse stored json or if none return initialValue
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      // If error also return initialValue
+      console.log(error);
+      return initialValue;
+    }
+  });
+  // Return a wrapped version of useState's setter function that ...
+  // ... persists the new value to localStorage.
+  const setValue: Setter<T> = (value, updateStorage) => {
```

Comment:
I think we can give `updateStorage` a default value of `true` (or even remove it) because we probably always want to persist this to the local storage. The overhead probably doesn't matter in most cases.